contracts/consensus-gateway/ConsensusCogateway.sol

```diff
@@ -94,6 +95,83 @@ contract ConsensusCogateway is MasterCopyNonUpgradable, MessageBus, ConsensusGat
         );
     }
 
+    /**
+     * @notice Confirm deposit in order to mint tokens.
+     *
+     * @param _amount MOST token deposit amount in atto.
+     * @param _beneficiary Address of beneficiary on auxiliary chain.
+     * @param _feeGasPrice Fee gas price at which rewards will be calculated.
+     * @param _feeGasLimit Fee gas limit at which rewards will be calculated.
+     * @param _depositor Address of depositor on the origin chain.
+     * @param _blockHeight Block height of origin chain at which storage proof
+     *                     is generated.
+     * @param _rlpParentNodes Storage merkle proof to verify message declaration
+     *                        on the origin chain.
+     *
+     * @return messageHash_ Message hash.
+     */
+    function confirmDeposit(
+        uint256 _amount,
+        address payable _beneficiary,
+        uint256 _feeGasPrice,
+        uint256 _feeGasLimit,
+        address _depositor,
+        uint256 _blockHeight,
+        bytes calldata _rlpParentNodes
+    )
+        external
+        returns (bytes32 messageHash_)
+    {
+
+        uint256 initialGas = gasleft();
+
+        require(
+            _amount != 0,
+            "Deposit amount should be greater than 0."
+        );
+        require(
+            _beneficiary != address(0),
+            "Beneficiary address must not be 0."
+        );
+        require(
+            _depositor != address(0),
+            "Depositor address must not be 0."
+        );
+
+        uint256 nonce = nonces[msg.sender];
+        nonces[msg.sender] = nonce.add(1);
+
+        messageHash_ = MessageInbox.confirmMessage(
+            ERC20GatewayBase.hashDepositIntent(
+                _amount,
+                _beneficiary
+            ),
+            nonce,
+            _feeGasPrice,
+            _feeGasLimit,
+            _depositor,
+            _blockHeight,
+            _rlpParentNodes
+        );
+
+        // Additional gas consumption after this statement can be adjusted with
+        // the gas price value.
+        uint256 gasConsumed = initialGas.sub(gasleft());
+        uint256 rewardAmount = reward(gasConsumed, _feeGasPrice, _feeGasLimit);
```

Comment:
perhaps `feeAmount` is a more pointed name?

Reply:
Done 👍